libs/go/aealite_agent_example/tests/pexpect_popen.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Wrapper for Pexpect to use in tests."""
+import os
+import platform
+import signal
+import sys
+import time
+from typing import List, Optional, Union
+
+from pexpect.exceptions import EOF, TIMEOUT  # type: ignore
+from pexpect.popen_spawn import PopenSpawn  # type: ignore
+
+from aea.helpers.base import send_control_c
+
+
+class PexpectWrapper(PopenSpawn):
+    """Utility class to make aea cli test easier."""
+
+    def __init__(self, *args, **kwargs):
+        """Init pexpect.spawn."""
+        if platform.system() != "Windows":
+            kwargs["preexec_fn"] = os.setsid
+
+        super().__init__(*args, **kwargs)
+
+    def control_c(self) -> None:
+        """Send control c to process started."""
+        time.sleep(0.1)  # sometimes it's better to wait a bit
+        send_control_c(self.proc, True)
+
+    @property
+    def returncode(self) -> Optional[Union[int, str]]:
+        """Get return code of finished process."""
+        return self.proc.poll()  # type: ignore
+
+    def wait_to_complete(self, timeout: float = 5) -> None:
+        """Wait process to complete.
+
+        Terminate automatically after timeout.
+        Set returncode to terminated if terminated.
+
+        :param timeout: how many seconds wait process to finish before kill.
+        """
+        if self.proc.poll() is not None:  # type: ignore
+            return
+
+        start_time = time.time()
+
+        while start_time + timeout > time.time() and self.proc.poll() is None:  # type: ignore
+            time.sleep(0.001)
```

Comment:
worth to make `0.001` customizable?

Reply:
does n make sense in this case, sleep done just for periodical check